iyinbor_lucky/calculator

LuckyIYI opened the pull request:
первая проверка калькулятора

LuckyIYI committed:
first commit

первая проверка калькулятора

NikAshanin commented:
Видимо не все файлы добавил, нет controller'a вообще среди запушенных файлов

LuckyIYI commented:
Да, проверил, случайно плохо скопировал в папку гихтаба, сейчас исправлю

> 27 нояб. 2017 г., в 23:21, Nikolai Ashanin <notifications@github.com> написал(а):
> 
> Видимо не все файлы добавил, нет controller'a вообще среди запушенных файлов
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/NikAshanin/HomeTasks/pull/39#issuecomment-347315310>, or mute the thread <https://github.com/notifications/unsubscribe-auth/Aarjo9J2qSpzCUlFEFYQQfFKhUv3zDiMks5s6xnLgaJpZM4Qq4-J>.
>

LuckyIYI commented:
Исправил, в это раз все должно работать

LuckyIYI committed:
second commit

добавил все файлы

NikAshanin commented on Calculator/.swiftlint.yml:
```diff
@@ -13,9 +13,7 @@ opt_in_rules:
-
-  - let_var_whitespace
-
+  
```
ненадо этот файл менять

NikAshanin commented on Calculator/.swiftlint.yml:
```diff
@@ -60,4 +58,4 @@ identifier_name:
 
-warning_threshold: 1
\ No newline at end of file
+warning_threshold: 2
```
ненадо этот файл менять

NikAshanin commented on Calculator/Calculator.xcodeproj/project.pbxproj:
```diff
@@ -145,21 +151,21 @@
 			outputPaths = (
 			);
-			runOnlyForDeploymentPostprocessing = 0;
+			runOnlyForDeploymentPostprocessing = 1;
```
возвращай обратно

NikAshanin replied:
??

NikAshanin commented on Calculator/Calculator/CalculatorViewController.swift:
```diff
@@ -1,5 +1,182 @@
 import UIKit
 
 final class CalculatorViewController: UIViewController {
+    private var userInTheMiddleOfTyping = false
```
isTyping? проще

NikAshanin commented on Calculator/Calculator/CalculatorViewController.swift:
```diff
@@ -1,5 +1,182 @@
+            dotAlreadySet = true
+        }
+    }
+    @IBAction func touchDigit(_ sender: UIButton) {
```
private

NikAshanin commented on Calculator/Calculator/CalculatorViewController.swift:
```diff
@@ -1,5 +1,182 @@
+            userInTheMiddleOfTyping = true
+        }
+    }
+    @IBAction func makeFuncChange(_ sender: Any) {
```
private

NikAshanin commented on Calculator/Calculator/CalculatorViewController.swift:
```diff
@@ -1,5 +1,182 @@
+            secondTurnOn = false
+        }
+    }
+    @IBAction func undoButton(_ sender: UIButton) {
```
private

NikAshanin commented on Calculator/Calculator/MathOperation.swift:
```diff
@@ -0,0 +1,213 @@
+import Foundation
+
+func factorialFunc(_ someValue: Int) -> Int {
```
делай extension к Int, вместо функций

NikAshanin commented on Calculator/Calculator/MathOperation.swift:
```diff
@@ -0,0 +1,213 @@
+    return facto
+}
+
+func degreesToRadians(_ number: Double) -> Double {
```
тоже самое

NikAshanin commented on Calculator/Calculator/MathOperation.swift:
```diff
@@ -0,0 +1,213 @@
+    var memoryArrayLength = 0
+    var buttonPressed = false
+    var result: Double? {
+            return accumulator
```
что-то уехало вправо

NikAshanin commented on Calculator/Calculator/MathOperation.swift:
```diff
@@ -0,0 +1,213 @@
+        memoryArrayLength = memoryArray.count - 1
+    }
+
+    mutating func checkPressedButton() {
```
сделай final class вместо структуры и убери все эти mutating func

NikAshanin commented on Calculator/Calculator/RoundButton.swift:
```diff
@@ -0,0 +1,9 @@
+@IBDesignable class RoundButton: UIButton {
+    @IBInspectable var cornerRadius: CGFloat = 0 {
+        didSet {
+            self.layer.cornerRadius = cornerRadius
```
self тут нужен?

NikAshanin commented on Calculator/Calculator/RoundButton.swift:
```diff
@@ -0,0 +1,9 @@
+import UIKit
+
+@IBDesignable class RoundButton: UIButton {
```
final

NikAshanin commented:
Фиксы будут?

LuckyIYI commented:
Да, конечно, сегодня сделаю, только  вернулся из поездки

Отправлено с iPhone

> 11 дек. 2017 г., в 14:41, Nikolai Ashanin <notifications@github.com> написал(а):
> 
> Фиксы будут?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

LuckyIYI committed:
third commit

Исправил все замечания. Не получается решить проблему со swiftlint правилом, отвечающим за пробелы. В случае когда пробел поставлен пишет одну ошбику, когда нет - другую

NikAshanin commented on Calculator/Calculator/MathOperation.swift:
```diff
@@ -0,0 +1,215 @@
+
+extension Int {
+    func factorialFunc() -> Int {
+        var facto = 1
```
тут же неверно написана логика

NikAshanin commented on Calculator/Calculator/MathOperation.swift:
```diff
@@ -0,0 +1,215 @@
+extension Int {
+    func factorialFunc() -> Int {
+        var facto = 1
+        if self < 2 {
```
если self < 0? всё ломается сразу
И тут нужно писать if self == 1 && self == 0

NikAshanin commented on Calculator/Calculator/MathOperation.swift:
```diff
@@ -0,0 +1,215 @@
+        if self < 2 {
+            return self
+        }
+        if self < 20 {
```
а если более 20? плохо написано тут. можешь возвращать тут ошибку или как то обрабатывать эту ситуацию вместо возвращения 1
